<a href="https://colab.research.google.com/github/fildzahazizi34rpl-jpg/Exploratory-Data-Analisis/blob/main/Praktikum_1_6.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
# DataFrame - seperti tabel Excel di dalam Python
df = pd.read_csv("data_kantin.csv")
df.head() # lihat 5 baris pertama
df.info() # lihat tipe data & jumlah data tiap kolom
df.describe() # lihat statistik ringkas (rata-rata, min, max, dst)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 110 entries, 0 to 109
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   tanggal   110 non-null    object 
 1   menu      107 non-null    object 
 2   kategori  110 non-null    object 
 3   harga     110 non-null    int64  
 4   terjual   104 non-null    float64
dtypes: float64(1), int64(1), object(3)
memory usage: 4.4+ KB


,harga,terjual
count,110.000000,104.000000
mean,6336.363636,22.923077
std,3520.001232,10.831178
min,2000.000000,8.000000
25%,3000.000000,13.000000
50%,5000.000000,20.000000
75%,10000.000000,33.000000
max,12000.000000,45.000000


In [ ]:
# NumPy Array Dasar

import numpy as np
nilai = np.array([80, 75, 90, 60, 88])
print(nilai.mean()) # rata-rata
print(nilai.max()) # nilai tertinggi
print(nilai.min()) # nilai terendah
print(nilai * 2) # operasi langsung ke semua elemen (vectorized)

78.6
90
60
[160 150 180 120 176]


In [ ]:
# Pandas Series as pd

import pandas as pd
data = {
'nama': ['Andi', 'Budi', 'Citra'],
'nilai': [80, 75, 90]
}
df = pd.DataFrame(data)
print(df)

    nama  nilai
0   Andi     80
1   Budi     75
2  Citra     90


In [ ]:
#Latihan 1

import numpy as np
harga = np.array([5000, 7000, 3000, 12000, 4500])
print('Rata-rata harga:', harga.mean())
print('Harga tertinggi:', harga.max())
print('Harga setelah diskon 10%:', harga * 0.9)

Rata-rata harga: 6300.0
Harga tertinggi: 12000
Harga setelah diskon 10%: [ 4500.  6300.  2700. 10800.  4050.]


Jawaban Tugas Analisis 1

Dari hasil program:

Rata-rata harga = Rp6.300
Harga tertinggi = Rp12.000
Harga setelah diskon 10% = Rp4.500, Rp6.300, Rp2.700, Rp10.800, dan Rp4.050

Jika menggunakan NumPy array, perintah harga * 0.9 akan mengalikan setiap elemen dengan 0,9 secara langsung.

In [ ]:
#Latihan 2

import pandas as pd
data_kantin = {
'menu': ['Nasi Goreng', 'Es Teh', 'Mie Ayam', 'Es Teh', None],
'harga': [12000, 4000, 10000, 4000, 8000],
'terjual': [23, 40, None, 35, 18]
}
df = pd.DataFrame(data_kantin)
print(df)

          menu  harga  terjual
0  Nasi Goreng  12000     23.0
1       Es Teh   4000     40.0
2     Mie Ayam  10000      NaN
3       Es Teh   4000     35.0
4         None   8000     18.0


Jika data langsung dianalisis tanpa dibersihkan, hasil analisis bisa kurang akurat atau menimbulkan kesalahan. Contohnya, data terjual yang kosong dapat memengaruhi perhitungan jumlah atau rata-rata penjualan, sedangkan menu yang kosong membuat kita tidak mengetahui nama menu yang terkait dengan data tersebut.

In [ ]:
#Latihan 3

df = pd.read_csv("data_kantin.csv")

print(df.head()) # 5 baris pertama
print(df.info()) # tipe data & jumlah non-null tiap kolom
print(df.describe()) # statistik ringkas kolom numerik
print(df.shape) # jumlah (baris, kolom)

      tanggal         menu kategori  harga  terjual
0  2026-08-20   Teh Hangat  Minuman   3000     11.0
1  2026-08-31  Nasi Goreng  Makanan  12000     29.0
2  2026-08-21  Nasi Goreng  Makanan  12000     28.0
3  2026-08-28   Teh Hangat  Minuman   3000     12.0
4  2026-08-21     Es Jeruk  Minuman   5000     19.0
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 110 entries, 0 to 109
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   tanggal   110 non-null    object 
 1   menu      107 non-null    object 
 2   kategori  110 non-null    object 
 3   harga     110 non-null    int64  
 4   terjual   104 non-null    float64
dtypes: float64(1), int64(1), object(3)
memory usage: 4.4+ KB
None
              harga     terjual
count    110.000000  104.000000
mean    6336.363636   22.923077
std     3520.001232   10.831178
min     2000.000000    8.000000
25%     3000.000000   13.000000
50%     5000.000000   20.000000
75%    10000.000000 

Kolom yang jumlah non-null-nya lebih sedikit dari jumlah baris total adalah menu dan terjual. Kolom menu memiliki 107 data terisi dari 110 baris, sehingga terdapat 3 data kosong. Kolom terjual memiliki 104 data terisi dari 110 baris, sehingga terdapat 6 data kosong. Artinya, terdapat beberapa data yang belum memiliki nilai dan perlu ditangani terlebih dahulu sebelum dilakukan analisis lebih lanjut agar hasilnya tetap akurat.

In [ ]:
#Latihan 4

print(df.isnull().sum()) # jumlah data kosong tiap kolom

df['terjual'] = df['terjual'].fillna(0) # isi kekosongan dengan 0

df = df.dropna(subset=['menu']) # hapus baris jika kolom menu kosong

tanggal     0
menu        3
kategori    0
harga       0
terjual     6
dtype: int64


Kolom terjual diisi dengan 0 menggunakan fillna(0) karena data kosong dapat dianggap sebagai tidak ada penjualan, sehingga baris tersebut masih bisa digunakan dalam analisis.

Sedangkan baris dengan menu kosong dihapus menggunakan dropna() karena nama menu merupakan informasi penting untuk mengetahui produk yang dijual. Jika nama menu kosong, data tersebut sulit dikelompokkan atau dianalisis berdasarkan menu.

Jadi, terjual diisi 0 agar data tetap dipertahankan, sedangkan menu yang kosong dihapus karena informasi utamanya tidak diketahui.

In [ ]:
#Latihan 5

print(df.duplicated().sum()) # jumlah baris duplikat
df = df.drop_duplicates()

df['harga'] = df['harga'].astype(int) # memastikan tipe data harga adalah integer
print(df.dtypes)

5
tanggal      object
menu         object
kategori     object
harga         int64
terjual     float64
dtype: object


Setelah menggunakan drop_duplicates(), jumlah baris dataset berkurang dari 110 baris menjadi 105 baris karena terdapat 5 baris duplikat. Memastikan tipe data (dtypes) sudah benar penting agar data dapat dianalisis dan dihitung dengan tepat. Contohnya, kolom harga harus berupa tipe angka (int64) agar dapat digunakan dalam operasi perhitungan seperti perkalian dan penjumlahan. Tipe data yang benar juga dapat mencegah kesalahan dalam proses analisis data.

In [ ]:
laris = df[df['terjual'] > 20] # filtering

urut = df.sort_values(by='terjual', ascending=False) # sorting

df['total_pendapatan'] = df['harga'] * df['terjual'] # kolom turunan

ringkasan = df.groupby('menu')['total_pendapatan'].sum() # agregasi
print(ringkasan)

menu
Es Jeruk       1460000.0
Es Teh         2144000.0
Kerupuk         972000.0
Mie Ayam       2410000.0
Nasi Goreng    3924000.0
Roti Bakar     1288000.0
Teh Hangat      459000.0
Name: total_pendapatan, dtype: float64


Berdasarkan hasil groupby, menu yang menghasilkan total_pendapatan tertinggi adalah Nasi Goreng, dengan total pendapatan sebesar Rp3.924.000.

Informasi ini dapat membantu kantin sekolah mengetahui menu yang memiliki pendapatan paling tinggi. Kantin dapat mempertahankan ketersediaan Nasi Goreng, menyiapkan stok yang cukup, dan menjadikannya salah satu menu yang diprioritaskan dalam penjualan. Dengan begitu, keputusan pengelolaan stok dan penjualan dapat dilakukan berdasarkan data.